compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -146,6 +147,7 @@ impl<'ll, 'tcx> ConstCodegenMethods<'tcx> for CodegenCx<'ll, 'tcx> {
     }
 
     fn const_int(&self, t: &'ll Type, i: i64) -> &'ll Value {
+        assert!(self.type_kind(t) == TypeKind::Integer, "only allows integer types in const_int");
```

Comment:
Debug asserts should be fine, I'm only hitting this while experimenting after all.